File: src/main/java/com/webtruyenapi/entity/Comic.java
package com.webtruyenapi.entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;
import com.fasterxml.jackson.annotation.JsonIgnore;

@Entity
@Table(name = "comics")
@Data
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Comic {
    @Id
    @Column(name = "comic_id", length = 36)
    private String comicId;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "slug", nullable = false)
    private String slug;

    @Column(name = "origin_name")
    private String originName;

    @Column(name = "status")
    @Enumerated(EnumType.STRING)
    private ComicStatus status;

    @Column(name = "thumb_url", columnDefinition = "TEXT")
    private String thumbUrl;

    @Column(name = "sub_docquyen")
    private Boolean subDocquyen = false;

    @Column(name = "chapters_latest", columnDefinition = "TEXT")
    private String chaptersLatest;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "modified_at")
    private LocalDateTime modifiedAt;

    @OneToMany(mappedBy = "comic", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Chapter> chapters = new ArrayList<>();

    @OneToMany(mappedBy = "comic", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<ComicGenre> comicGenres = new ArrayList<>();

    @OneToMany(mappedBy = "comic", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnore
    private List<ComicFollow> comicFollows;

    @PrePersist
    protected void onCreate() {
        if (this.createdAt == null) {
            this.createdAt = LocalDateTime.now();
        }
        if (this.updatedAt == null) {
            this.updatedAt = LocalDateTime.now();
        }
        if (this.subDocquyen == null) {
            this.subDocquyen = false;
        }
    }

    @PreUpdate
    protected void onUpdate() {
        this.modifiedAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }
}

File: Models/Comic.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using webtruyenBackEndAPI.Models;

namespace webtruyenBackEndAPI.Models
{

	[Table("comics")]
	public class Comic
	{
		[Key]
		[Column("comic_id")]
		public string? ComicId { get; set; }

		[Column("name")]
		public string Name { get; set; }

		[Column("slug")]
		public string Slug { get; set; }

		[Column("origin_name")]
		public string? OriginName { get; set; }

		[Column("status")]
		public string? Status { get; set; }

		[Column("thumb_url")]
		public string? ThumbUrl { get; set; }

		[Column("sub_docquyen")]
		public bool SubDocquyen { get; set; }

		[Column("chapters_latest")]
		public string? ChaptersLatest { get; set; }

		[Column("updated_at")]
		public DateTime? UpdatedAt { get; set; }

		[Column("created_at")]
		public DateTime? CreatedAt { get; set; }

		[Column("modified_at")]
		public DateTime? ModifiedAt { get; set; }

		// 🔗 Quan hệ
		public ICollection<Chapter> Chapters { get; set; }
		public ICollection<ComicGenre> ComicGenres { get; set; }
	}
}
